src/v/cluster/partition.cc

```diff
@@ -532,41 +534,41 @@ partition::timequery(storage::timequery_config cfg) {
 }
 
 bool partition::may_read_from_cloud() const {
-    return _cloud_storage_partition
-           && _cloud_storage_partition->is_data_available();
+    return is_remote_fetch_enabled()
```

Comment:
Could the fix functionally have been adding this one line? Just making sure I'm not missing any other functional changes.

In the future, it'd be great if you could split functional and non-functional changes into separate commits, so it's easier for reviewers to tell where to spend more cycles looking.

Reply:
>Could the fix functionally have been adding this one line? Just making sure I'm not missing any other functional changes.

@WillemKauf just bumping this first question, in case there's anything i'm missing

Reply:
Thanks!